### PR TITLE
docs(redis sink): Use generated docs for configuration

### LINKED ISSUE #16326: Convert the `redis` sink to use auto-generated docs

### PR DESCRIPTION
**Pull request by StephenWakely**
Closes #16326 
Epic #14999 

Signed-off-by: Stephen Wakely <fungus.humungus@gmail.com>

### CODE REVIEW

File: src/sinks/redis.rs [hunk 1]
```diff
@@ -124,14 +124,17 @@ pub struct RedisSinkConfig {
...
 
-    /// The Redis URL to connect to.
+    /// The URL of the Redis endpoint to connect to.
     ///
     /// The URL _must_ take the form of `protocol://server:port/db` where the protocol can either be
     /// `redis` or `rediss` for connections secured via TLS.
-    url: String,
+    #[configurable(metadata(docs::examples = "redis://127.0.0.1:6379/0"))]
+    #[serde(alias = "url")]
+    endpoint: String,
```
Comment (spencergilbert) [inline src/sinks/redis.rs:133]: ❤️ love this change, but I think it also needs a note in the upcoming upgrade guide?

**Review by spencergilbert — CHANGES_REQUESTED** [review]

File: src/sinks/redis.rs (hunk 1 above)
Comment (StephenWakely) [reply src/sinks/redis.rs:133]: Done.

File: website/content/en/highlights/2023-02-28-0-28-0-upgrade-guide.md [hunk 2]
```diff
@@ -12,6 +12,7 @@ badges:
 Vector's 0.28.0 release includes **breaking changes**:
 
 1. [The `journald` source's `units` option removed](#journald-units)
+2. [The `redis` sink `url` argument changed to `endpoint`](#redis-endpoint)
```
Comment (spencergilbert) [inline website/content/en/highlights/2023-02-28-0-28-0-upgrade-guide.md]:
```suggestion

and **deprecations**:

1. [The `redis` sink `url` argument changed to `endpoint`](#redis-endpoint)
```

Somewhat pedantic but more in line with how we've announced these in the past.

File: website/content/en/highlights/2023-02-28-0-28-0-upgrade-guide.md [hunk 3]
```diff
@@ -23,3 +24,9 @@ We cover them below to help you upgrade quickly:
 
 The `units` option had been replaced by `include_units` in the `0.10` release. This release marks the final removal
 of this option, please update your configurations to use the `include_units` option instead.
+
+#### The `redis` sink `url` argument changed to `endpoint` {#redis-endpoint}
```
Comment (spencergilbert) [inline website/content/en/highlights/2023-02-28-0-28-0-upgrade-guide.md]:
```suggestion

### Deprecation notices

#### The `redis` sink `url` argument changed to `endpoint` {#redis-endpoint}
```

**Review by spencergilbert — CHANGES_REQUESTED** [review]
Slight tweak on the highlights to be more aligned with how we've announced deprecations in the past ([ref](https://vector.dev/highlights/2023-01-17-0-27-0-upgrade-guide/))

**Review by spencergilbert — APPROVED** [review]

**Review by neuronull — APPROVED** [review]

**Pull request merged**